lib/mayaUsd/utils/diagnosticDelegate.cpp

```diff
@@ -107,7 +107,7 @@ void UsdMayaDiagnosticDelegate::IssueError(const TfError& err)
     // In addition, always display the full call site for errors by going
     // through _FormatDiagnostic.
     if (ArchIsMainThread()) {
-        MGlobal::displayError(_FormatDiagnostic(err));
+        MGlobal::displayError(err.GetCommentary().c_str());
```

Comment:
Why not do a fix inside of _FormatDiagnostic? If we fall, into the else then it will still print all the info (function/line/file) for error/status/warning. How about inside of _FormatDiagnostic you use if DEBUG check to print all info and in release only the commentary?

Reply:
@seando-adsk Yeah, I thought about doing it that way but then not all the developers work in DEBUG. :) I myself most of the time work in RelWithDebInfo and barely ever bother with Debug.

The only other option would be to introduce an ENV variable to control this behaviour but again this could be controversial.

Reply:
But at least having it in DEBUG would provide an option to show all the info (file/function/line). With your change the only way to get that is to not be in the main thread, which I don't know how/why we wouldn't be.

Reply:
Sure, I don't have a strong preference but do people really need this? Why stop at "Errors"? What about Warnings? 

For example, I see in many places in our code based people tend to use TF_WARN when checking for object validation.

```
if (shaderInstance == nullptr) {
    TF_WARN("Failed to create shader instance for %s", nodeId.asChar());
    break;
}
```